NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# Ja edit post 8

Description
As an author, I would like the ability to edit a post I have written so that I can change details that are no longer relevant or correct mistakes.

Given an author is viewing a specific Post that they have written 
When they select the edit option for a Post they have authored
Then the user should be directed to a form and given the ability to change the Post’s content

Given the author wishes to confirm the edits
When they the select the option to confirm
Then the Post saved to the database via UPDATE
And the author should be directed back to the Post Details for this Post

Given the author decides not to confirm the Edit
When they the select the option to reject confirmation
Then the Post should NOT show any Edits
And the author should be directed back to the MyPost list
Related Ticket(s)  

#8 - https://trello.com/c/1RnwCV3Z/8-8-edit-a-post
Type of Change
☑︎ New feature (non-breaking change which adds functionality)
Testing Instructions
#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
• Clone this project’s repository to your machine
   
   • If you’ve already done this and are currently working on a different branch of the project, be sure to git add and git commit whatever you’re working on on your branch, before you attempt to switch branches
   • Then git checkout main to switch to the project’s main branch
• Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
• git fetch -a to retrieve the current remote-tracked branches for the project, you should see a branch named JA-editPost-8 as one of the available branches
• git checkout JA-editPost-8 to switch to my branch
• If you don't see the most recent version of my files, git pull origin JA-editPost-8
• Start Tabloid.sln to open the project in Visual Studio
• If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
   
   • Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
   • ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)
• You will be presented with a login screen. Enter the email [foo@bar.com](mailto:foo@bar.com) and click "Login" (no password required)
• You will be presented with the welcome screen that simply says "hello"
• Click the  MPOST link in the topside header
-You should see a table that contains all my posts from your database as well as their corresponding Id. This table should be in  order by date created
-To the bottom of each item on the list you should see an details link
-Click said link 
-You will then be redirected to a details page with the yellow edit button on the bottom

- edit the fields and click save
 
SELECT * FROM Post;
to confirm that your post has been edited to the database.
Checklist:
☑︎ My code follows the style guidelines of this project
☑︎ I have performed a self-review of my code
☑︎ I have commented my code, particularly in hard-to-understand areas
☑︎ My changes generate no new warnings or errors
☑︎ I have added test instructions that prove my fix is effective or that my feature works
